swift-generator/src/main/resources/templates/java/common.st

```diff
@@ -333,3 +336,47 @@ public String toString()
         .toString();
 }
 >>
+
+_union_constructor(context) ::= <<
+
+private Object value;
+private int id = -1;
+private String name;
+
+public <context.javaName>() {
```

Comment:
If possible, let's be more clear about what a valid state is for this object. If I understand correctly, this constructor allows a "nothing is set" instance to be created, where before this commit it's not possible to create a "nothing is set" instance.

In terms of consistency within this commit, the private constructor suggests that a valid state is `name != null`, but this constructor doesn't have that requirement. 

What's the real constraint we want to uphold?